extensions/amp-consent/0.1/consent-state-manager.js

```diff
@@ -34,9 +34,25 @@ import {assertHttpsUrl} from '../../../src/url';
 import {dev, devAssert} from '../../../src/log';
 import {expandConsentEndpointUrl, getConsentCID} from './consent-config';
 import {hasOwn} from '#core/types/object';
+import {once} from '#core/types/function';
+import {getRandomString64} from '#service/cid-impl';
+import {getAmpdoc, getServicePromise} from '../../../src/service-helpers';
 
 const TAG = 'CONSENT-STATE-MANAGER';
 
+/**
+ * Returns a promise for a service for the given id and ampdoc. Also expects
+ * a service that has the actual implementation. The promise resolves when
+ * the implementation loaded.
+ * @param {!Element|!ShadowRoot|!./service/ampdoc-impl.AmpDoc} element
+ * @return {!Promise<!Object>}
+ */
+export function getConsentStateManager(element) {
+  const ampdoc = getAmpdoc(element);
+  const isSingleDoc = ampdoc.isSingleDoc() ? ampdoc.win : ampdoc;
+  return getServicePromise(isSingleDoc, 'consentStateManager');
```

Comment:
You don't need to resolve the AmpDoc, only pass the element.
```suggestion
  return getServicePromise(element, 'consentStateManager');
```

Please also update `consent-ui.js` to use this helper here:

https://github.com/ampproject/amphtml/blob/bff81cb205f8bfde03123d1e29511dfd58b145ba/extensions/amp-consent/0.1/consent-ui.js#L529-L532